pkgs/misc/dart-sass-embedded/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, stdenvNoCC
+, fetchurl
+, autoPatchelfHook
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "dart-sass-embedded";
+  version = "1.62.1";
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = lib.optional stdenvNoCC.hostPlatform.isLinux autoPatchelfHook;
+
+  src = let base = "https://github.com/sass/dart-sass-embedded/releases/download/${version}/sass_embedded-${version}"; in
```

Comment:
Why not compile from source?

Reply:
dart programs can't be compiled from source in nixpkgs yet, as far as I am aware

Reply:
It doesn't look like there's an explicit builder for it, but I think it would work to just use the `dart` compiler in a plain stdenv environment with [fetch-dart-deps](https://github.com/NixOS/nixpkgs/blob/b7b2b2ad28d17da94727b364507f772ea9366791/pkgs/build-support/dart/fetch-dart-deps/default.nix) to grab the dependencies.

The build instructions look pretty straightforward: https://github.com/sass/dart-sass#from-source.

I don't think it's the end of the world to patch binaries like this (especially if blocking other things), but building from source would be ideal.

Reply:
Oh oops, sorry gave the link for the non-embedded repo

Reply:
:cry:  I tried to build it from source with:

```nix
{ lib
, stdenvNoCC
, fetchFromGitHub
, buf
, dart
, callPackage
}:

stdenvNoCC.mkDerivation (finalAttrs: {
  pname = "dart-sass-embedded";
  version = "1.62.1";

  src = fetchFromGitHub {
    owner = "sass";
    repo = "dart-sass-embedded";
    rev = "refs/tags/${finalAttrs.version}";
    hash = "sha256-GpSus5/QItbzCrOImMvrO6DTAQeODABRNiSYHJlLlIA=";
  };

  nativeBuildInputs = [
    buf
    dart
    (callPackage ../../build-support/dart/fetch-dart-deps {  } {
      buildDrvArgs = finalAttrs;
      vendorHash = "sha256-aEBE+z8M5ivMR9zL7kleBJ8c9T+4PGXoec56iwHVT+c=";
    })
  ];

  preBuild = ''
    runHook preBuild
    dart run grinder protobuf
    dart run grinder pkg-standalone-dev
    runHook postBuild
  '';

  installPhase = ''
    ls -la
  '';

  meta = with lib; {
    description = "A wrapper for Dart Sass that implements the compiler side of the Embedded Sass protocol";
    homepage = "https://github.com/sass/dart-sass-embedded";
    changelog = "https://github.com/sass/dart-sass-embedded/blob/${version}/CHANGELOG.md";
    license = licenses.mit;
    maintainers = with maintainers; [ shyim ];
  };
})
```

but dart just ended up segfaulting:
```
dart-sass-embedded> unpacking sources
dart-sass-embedded> unpacking source archive /nix/store/1q84s8r7a16mw1l1gwimqb6vrlzicq8p-source
dart-sass-embedded> source root is source
dart-sass-embedded> patching sources
dart-sass-embedded> configuring
dart-sass-embedded> no configure script, doing nothing
dart-sass-embedded> building
error: builder for '/nix/store/b9aiwfh5lfn63bzc47cil6dji562x6z7-dart-sass-embedded-1.62.1.drv' failed due to signal 11 (Segmentation fault);
       last 7 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/1q84s8r7a16mw1l1gwimqb6vrlzicq8p-source
       > source root is source
       > patching sources
       > configuring
       > no configure script, doing nothing
       > building
       For full logs, run 'nix log /nix/store/b9aiwfh5lfn63bzc47cil6dji562x6z7-dart-sass-embedded-1.62.1.drv'.
```

so I think for now a binary package would be ok.